### PR TITLE
Update project structure

### PR DESCRIPTION
**Pull request by FrankHossfeld**

### CODE REVIEW

File: gwt-dom-gwt2-tests/src/test/java/org/gwtproject/dom/builder/client/GwtParagraphBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  @Override
+  public String getModuleName() {
+    return GWT_MODULE_NAME;
+  }
+
+  /** Test that HTML can be set after ending one element and starting another. */
+  @Override
+  public void testHtmlAfterRestart() {
+    /*
+     * This test triggers an obscure bug in IE7 where you cannot set the
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-dom-gwt2-tests/src/test/java/org/gwtproject/dom/builder/client/GwtParagraphBuilderTest.java:32]: Removed test class as we do not support IE7

File: gwt-dom-gwt2-tests/src/test/java/org/gwtproject/dom/client/TableTests.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    assertEquals(cell2, row.getCells().getItem(1));
+    assertEquals(cell3, row.getCells().getItem(2));
+
+    TableCellElement cell0 = row.insertCell(0);
+    assertEquals(cell0, row.getCells().getItem(0));
+    assertEquals(cell1, row.getCells().getItem(1));
+    assertEquals(cell2, row.getCells().getItem(2));
+    assertEquals(cell3, row.getCells().getItem(3));
+
+    // TODO: TableCellElement.cellIndex is broken (always 0) on Safari 2 (bug 3295)
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-dom-gwt2-tests/src/test/java/org/gwtproject/dom/client/TableTests.java:80]: Fixed in Safari 3

File: gwt-dom/src/main/resources/org/gwtproject/media/dom/MediaDOM.gwt.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<module>
+  <!-- sources -->
+  <source path="client"/>
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-dom/src/main/resources/org/gwtproject/media/dom/MediaDOM.gwt.xml:20]: No, no dependency to dom. 
You mean I can remove `<source path="client"/>`?

[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-dom/src/main/resources/org/gwtproject/media/dom/MediaDOM.gwt.xml:20]: Think, it is required

File: gwt-dom/src/main/resources/org/gwtproject/dom/DOM.gwt.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  <inherits name="org.gwtproject.core.Core"/>
+  <inherits name="org.gwtproject.canvas.dom.CanvasDOM"/>
+  <inherits name="org.gwtproject.dom.builder.BuilderDOM" />
+  <inherits name="org.gwtproject.dom.style.Style" />
+  <inherits name="org.gwtproject.media.dom.MediaDOM"/>
+  <inherits name="org.gwtproject.safehtml.SafeHtml"/>
+
+  <!-- sources -->
+  <source path="client" />
+  <public path="public" />
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-dom/src/main/resources/org/gwtproject/dom/DOM.gwt.xml:42]: remove public, keep client, cause test fails if I remove it.

File: gwt-dom/src/main/resources/org/gwtproject/canvas/dom/CanvasDOM.gwt.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    limitations under the License.
+
+-->
+<module>
+  <!-- Other module inherits                                      -->
+  <inherits name="org.gwtproject.dom.DOM"/>
+
+  <!-- sources -->
+  <source path="client"/>
+  <public path="public" />
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-dom/src/main/resources/org/gwtproject/canvas/dom/CanvasDOM.gwt.xml:24]: removed `public`, keep `client`

File: gwt-dom/src/main/resources/org/gwtproject/media/dom/MediaDOM.gwt.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-dom/src/main/resources/org/gwtproject/media/dom/MediaDOM.gwt.xml:20]: It seams not necessary, but test fails. I will remove only the wrong references and keep

File: gwt-dom-gwt2-tests/src/test/java/org/gwtproject/dom/builder/client/GwtParagraphBuilderTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (niloc132) [reply gwt-dom-gwt2-tests/src/test/java/org/gwtproject/dom/builder/client/GwtParagraphBuilderTest.java:32]: i think you just want to remove the override, and leave the class? the override exists just stops a test from happening

File: gwt-dom/src/main/resources/org/gwtproject/media/dom/MediaDOM.gwt.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (niloc132) [reply gwt-dom/src/main/resources/org/gwtproject/media/dom/MediaDOM.gwt.xml:20]: Ok, we can resolve later, or just ignore it, as long as it works.

[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply gwt-dom/src/main/resources/org/gwtproject/media/dom/MediaDOM.gwt.xml:20]: Let's keep it.

**Pull request merged**